api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImpl.java

```diff
@@ -41,8 +43,8 @@
 	
 	@Override
 	@Transactional(readOnly = true)
-	public Patient getPatientByUuid(String uuid) {
-		return translator.toFhirResource(dao.getPatientByUuid(uuid));
+	public Patient get(String uuid) {
+		return translator.toFhirResource(dao.get(uuid));
```

Comment:
Technically, you should be able to remove this entire method as it's implemented in `BaseFhirService` now.

Reply:
hi @ian  did you mean the get() for this(String uuid)?

Reply:
@ibacher hei ian,what do you think of classess like FhirPatientDaoImpl,it extends BasePersonDao and makes use of its methods,are we dropping it,if so what about some of its methods that we are using??thanks

Reply:
I don't think that you need to drop `BasePersonDao` obviously we need methods like `searchForPatients` etc. I guess @ibacher meant to say that you should remove `getPatientByUuid` method from Dao as well as from the Service class as the `get` method is already implemented in the `BaseFhirService`

Reply:
Yes, what @VaishSiddharth said. The simple get, create, update, delete methods should all be implemented by BaseFhirDao. The addition stuff shared among person objects (Person, Provider, Patient, User) should still live in BasePersonDao.

Reply:
sicne FhirPatientDaoImpl extends BasePersonDao arleady,i am wondering on the work round to have the two extended ie plus BaseFhirDao. cc  @ibacher @VaishSiddharth

Reply:
@HerbertYiga So if you see closely in `BasePersonDao` it already extends `BaseFhirDao` so no need to extend that class again. And I don't think we can actually extend two classes.

Reply:
No, you can only extend one class at a time in Java... the alternative is multiple inheritance, which [can get messy](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem). That said, the class hierarchy can be as deep as you want so `FhirPatientDaoImpl` -> `BasePersonDao` -> `BaseFhirDao` -> `BaseDao` is fine. That way, `FhirPatientDaoImpl` has all the functionality of `BaseDao` and the functionality of `BaseFhirDao` and the functionality of `BasePersonDao`.